### PR TITLE
Allow users to set a remote certificate validation callback for the IoT hub service client

### PR DESCRIPTION
**Pull request by timtay-microsoft**
currently supported over HTTP and AMQP TCP. Will add AMQP WS support later

### CODE REVIEW

File: iothub/service/src/IotHubServiceClientOptions.cs [hunk 1]
```diff
@@ -101,5 +103,28 @@ public class IotHubServiceClientOptions
         /// </para>
         /// </remarks>
         public TimeSpan AmqpConnectionKeepAlive { get; set; } = TimeSpan.FromMinutes(2);
+
+        /// <summary>
+        /// A callback for remote certificate validation.
+        /// </summary>
+        /// <remarks>
+        /// If incorrectly implemented, your device may fail to connect to IoT Hub and/or be open to security vulnerabilities.
```
Comment (drwill-ms) [inline iothub/service/src/IotHubServiceClientOptions.cs:111]:
```suggestion
        /// If incorrectly implemented, your device may fail to connect to IoT hub and/or be open to security vulnerabilities.
```

File: iothub/service/src/IotHubServiceClientOptions.cs [hunk 2]
```diff
@@ -101,5 +103,28 @@ public class IotHubServiceClientOptions
...
+        /// <para>
+        /// This feature is only applicable for HTTP connections and for AMQP TCP connections. AMQP web socket communication
+        /// does not support this feature.
+        /// </para>
+        /// </remarks>
+        public RemoteCertificateValidationCallback RemoteCertificateValidationCallback { get; set; } = DefaultRemoteCertificateValidation;
```
Comment (drwill-ms) [inline iothub/service/src/IotHubServiceClientOptions.cs:117]: Does this apply to both device and module client?

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/IotHubServiceClientOptions.cs (hunk 2 above)
Comment (timtay-microsoft) [reply iothub/service/src/IotHubServiceClientOptions.cs:117]: Yes, but MQTT also supports this in device/module client

**Review by abhipsaMisra — APPROVED** [review]

**Pull request merged**